### PR TITLE
Added support for 13/19 digits VISA credit cards in PaymentCardNumber type

### LINKED ISSUE #1416: Support for 13/19 digits VISA credit cards in PaymentCardNumber type
# Feature Request

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.5
            pydantic compiled: True
                 install path: /Users/<hidden>/.pyenv/versions/3.7.2/envs/common-utils/lib/python3.7/site-packages/pydantic
               python version: 3.7.2 (default, Jan 22 2020, 18:22:02)  [Clang 10.0.1 (clang-1001.0.46.4)]
                     platform: Darwin-18.7.0-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions']
```

Currently `PaymentCardNumber` type does support 16-digits VISA only:

https://github.com/samuelcolvin/pydantic/blob/3cd8b1ee2d5aac76528dbe627f40fe1c27bf59f6/pydantic/types.py#L677-L679

However as described on wiki page https://en.wikipedia.org/wiki/Payment_card_number:

> While the vast majority of Visa's account ranges describe 16 digit card numbers there are still a few account ranges (forty as of 11 December 2013) dedicated to 13 digit PANs and several (439 as of 11 Dec. 2013) account ranges where the issuer can mix 13 and 16 digit card numbers. Visa's VPay brand can specify PAN lengths from **13** to **19** digits and so card numbers of more than 16 digits are now being seen.

This fact is handler properly in `faker` library:

https://github.com/joke2k/faker/blob/0f86b60178fbbd27d50a184e94c429d4b60d8aef/faker/providers/credit_card/__init__.py#L48-L49

But not handled in pydantic.

### PR DESCRIPTION
**Pull request by AlexanderSov**
## Change Summary

Added support for 13/19 digits VISA credit cards in PaymentCardNumber type.

## Related issue number

fix #1416

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: tests/test_types_payment_card_number.py [hunk 1]
```diff
@@ -9,7 +9,9 @@
 
 VALID_AMEX = '370000000000002'
 VALID_MC = '5100000000000003'
-VALID_VISA = '4050000000000001'
+VALID_VISA_16 = '4050000000000001'
+VALID_VISA_13 = '4050000000001'
+VALID_VISA_19 = '4050000000001'
```
Comment (StephenBrown2) [inline tests/test_types_payment_card_number.py:14]: This should probably be 19 characters, and sorted by length...
```suggestion
VALID_VISA_13 = '4050000000001'
VALID_VISA_16 = '4050000000000001'
VALID_VISA_19 = '4050000000000000001'
```

Comment (AlexanderSov) [reply tests/test_types_payment_card_number.py]: Yes, thank you. Corrected

File: changes/1416-AlexanderSov.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Added support for 13/19 digits VISA credit cards in PaymentCardNumber type
```
Comment (samuelcolvin) [inline changes/1416-AlexanderSov.md:1]:
```suggestion
Added support for 13/19 digits VISA credit cards in `PaymentCardNumber` type
```

File: pydantic/types.py [hunk 3]
```diff
@@ -799,10 +799,13 @@ def validate_length_for_brand(cls, card_number: 'PaymentCardNumber') -> 'Payment
         Validate length based on BIN for major brands:
         https://en.wikipedia.org/wiki/Payment_card_number#Issuer_identification_number_(IIN)
         """
-        required_length: Optional[int] = None
-        if card_number.brand in {PaymentCardBrand.visa, PaymentCardBrand.mastercard}:
+        required_length: Optional[Union[int, str]] = None
```
Comment (samuelcolvin) [inline pydantic/types.py:802]:
```suggestion
        required_length: Union[None, int, str] = None
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise looks great.

**Comment by AlexanderSov** [comment]
@samuelcolvin Can we merge this PR?

**Comment by samuelcolvin** [comment]
thanks so much. 🎉

**Pull request merged**